### ISSUE TITLE
Media not displaying when simulating a stream

### ISSUE THREAD

**Issue by succinct0274** [OvidijusParsiunas/deep-chat#45]
Hi Ovidijus, first of all, thank you for the great work you have made. It is a really nice project. I recently took this to make a chatbot with Langchain. I noticed that images will not be shown if simulated stream is enabled. I have spent some time looking at the source code. It seems that the static method `Stream.simulate` will not consider any images in the response. Will there be any future enhancement on this?

Anyway, thanks in advance.

**Comment by OvidijusParsiunas** [comment]
Hi @succinct0274.
Currently streaming supports text only. I have been thinking about improving this API to allow other data types. Could you tell me more about your use-case and would the stream with an image look in the chat:

Would you want an image to appear alongside text in the same bubble? So there would be streamed text, then an image would appear, then potentially more streamed text after it?
Or perhaps would you want the image to be in a different bubble?
Would there be multiple images?

Thanks!

**Comment by succinct0274** [comment]
Hi @OvidijusParsiunas, my backend server will plot the chart and send back as the response along with the text, the response format would be like
```{json}
{
  "text": <text response from the open ai>,
  "images": [ { "content": <binary image encoded into base64 string>, "content_type": <mime_type string>},  ...]
} 
```
the handler at the frontend side will receive the response and further convert into the type `MessageContent`, which is specified in the documentation.

**Comment by OvidijusParsiunas** [comment]
Ohhhh, are you referring to a stream simulation here? Not an actual stream with consecutive text updates?

**Comment by succinct0274** [comment]
yes, I am mentioning the simulated stream

**Comment by OvidijusParsiunas** [comment]
Sorry I misread the issue. I should be able to fix this.
In terms of user experiences I think the media/images should appear first and the text second so as-to not constantly move the image down as the text is populated. Will this work for you?

**Comment by succinct0274** [comment]
Sure, as long as images are s

**Comment by succinct0274** [comment]
Sure, thanks for your help

**Comment by OvidijusParsiunas** [comment]
I have written the initial code for this and will be able to bundle it with the next release that is going to happen in the next couple of days. I will update this thread on any news.

**Comment by OvidijusParsiunas** [comment]
Hey, sorry for the hold up as there have been a lot of changes happening in the component code. This feature will be released in the main package later this week, but you can use it early in our `deep-chat-dev` and `deep-chat-react-dev` packages version `9.0.66`. These packages work the same way it's just their names are different. Let me know if you have any issues. I will update this thread when the main package is released.

**Comment by OvidijusParsiunas** [comment]
Hi. The functionality discussed above has now been released in Deep Chat version `1.4.4`. Please read the [release notes](https://github.com/OvidijusParsiunas/deep-chat/releases/tag/1.4.4) for more information.

To note, `simulation` functionality has been changed to populate by character rather than word. So the recommended default number value is _6_, but you can ofcourse change it to your preference.

I will be closing this issue, but if you experience problems relating to streaming feel free to comment below or raise a new issue for anything else. Thanks!
